src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,217 @@
+import type { ReactElement, SetStateAction, TouchEventHandler } from 'react'
+import { useEffect, useState } from 'react'
+import { ICON } from '@constants/icons'
+
+export interface CarouselProps {
+  images: { url: string; id: number }[]
+  isArrow: boolean
+}
+
+interface SliderProps {
+  cursorOn: boolean
+}
+
+interface CurrentDotProps {
+  imageIndex: number
+}
+
+const Carousel = ({ images, isArrow }: CarouselProps): ReactElement => {
+  const carouselWidthSize = 60
+  const arrowRight = ICON.CHEVRON_RIGHT_40
+  const arrowLeft = ICON.CHEVRON_LEFT_40
+  const [translateValue, setTranslateValue] = useState<number>(0)
+  const [startClientX, setStartClientX] = useState<number>(0)
+  const [endClientX, setEndClientX] = useState<number>(0)
+  const [cursorOn, setCursorOn] = useState<boolean>(false)
+
+  const handleMoveCurrent = (imageIndex: number): void => {
+    setTranslateValue((imageIndex - 1) * carouselWidthSize)
+  }
+  const handleMoveRight = (): void => {
+    if (translateValue !== carouselWidthSize * (images.length - 1)) {
+      setTranslateValue(prev => prev + carouselWidthSize)
+    } else {
+      setTranslateValue(0)
+    }
+  }
+
+  const handleMoveLeft = (): void => {
+    if (translateValue !== 0) {
+      setTranslateValue(prev => prev - carouselWidthSize)
+    } else {
+      setTranslateValue(carouselWidthSize * (images.length - 1))
+    }
+  }
+
+  const handleMouseDown = (e: { clientX: SetStateAction<number> }): void => {
+    setStartClientX(e.clientX)
+    setCursorOn(true)
+  }
+
+  const handleMouseUp = (e: { clientX: SetStateAction<number> }): void => {
+    setEndClientX(e.clientX)
+    setCursorOn(false)
+  }
+  const handleTouchStart: TouchEventHandler<HTMLDivElement> = e => {
+    setEndClientX(e.touches[0].clientX)
+    setCursorOn(true)
+  }
+  const handleTouchStartEnd: TouchEventHandler<HTMLDivElement> = e => {
+    setStartClientX(e.changedTouches[0].clientX)
+    setCursorOn(false)
+  }
+
+  useEffect(() => {
+    const dragSpace = Math.abs(startClientX - endClientX)
+    const userSlideRight = endClientX < startClientX && dragSpace > 100
+    const userSlideLeft = endClientX > startClientX && dragSpace > 100
+    if (startClientX === 0) {
+      return
+    }
+    if (userSlideRight) {
+      handleMoveRight()
+    } else if (userSlideLeft) {
+      handleMoveLeft()
+    }
+  }, [endClientX])
+
+  return (
+    <StyledCarouselWrapper>
+      <StyledSlider
+        cursorOn={cursorOn}
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
+        onTouchEnd={handleTouchStartEnd}
+        onTouchStart={handleTouchStart}>
+        <StyledImageBox translateValue={translateValue || null}>
+          {images.map((image, idx) => {
+            return (
+              <StyledImage key={image.id} alt={'image' + idx} src={image.url} />
```

Comment:
alt값을 변경이 될 수 있는 idx보다 고유의 식별자가 될 수 있는 id를 조합하는 방식으로 변경하는건 어떨까요 ?
케러셀 이미지를 알려주는 네이밍으로 백틱을 사용해서 변경하면 더 좋을 것 같습니다!
alt는 이미지를 문자로 접하는 사용자에게는 이미지에 대한 정보를 제공하는 용도로도 사용되니
이미지에 대한 의미를 알려드릴 수 있는 prop을 받아 사용하면 좋을 것 같아요

```suggestion
              <StyledImage key={image.id} alt={`${name}-${id}`} src={image.url} />
```

Reply:
위 코멘트와 동일한 이유입니닷 ㅠ

Reply:
엇 제가 생각한 name은 이미지 각각의 name이 아닌 이 carousel의 이름을 생각했어요 
예시로 상세페이지의 carousel인 경우에는 production이라는 name을 받아서 
해당 carousel이 어떤 포지션으로 사용되는지를 얘기해주는 것이요!

정리하자면, 이미지 각각의 name이 아닌 carousel의 named을 받는다고 생각하시면 될 것 같습니다 ! ㅎㅎ

Reply:
홍홍 민망하네용~ 반영하겠습니닷! 귣귣